.github/workflows/solidity-foundry.yml

```diff
@@ -11,28 +11,55 @@ jobs:
     name: Detect changes
     runs-on: ubuntu-latest
     outputs:
-      changes: ${{ steps.changes.outputs.src }}
+      src_changes: ${{ steps.changes.outputs.src }}
+      sol_changes: ${{ steps.changes.outputs.sol }}
+      sol_files: ${{ steps.changes.outputs.sol_files }}
     steps:
       - name: Checkout the repo
         uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
       - uses: dorny/paths-filter@de90cc6fb38fc0963ad72b210f1f284cd68cea36 # v3.0.2
         id: changes
         with:
+          list-files: 'shell'
           filters: |
             src:
               - 'contracts/src/v0.8/**/*'
               - '.github/workflows/solidity-foundry.yml'
               - 'contracts/foundry.toml'
               - 'contracts/gas-snapshots/*.gas-snapshot'
+            sol:
+              - modified|added: 'contracts/src/v0.8/**/*.sol'
 
   tests:
+    if: needs.changes.outputs.src_changes == 'true'
     strategy:
       fail-fast: false
       matrix:
-        product: [automation, ccip, functions, keystone, l2ep, liquiditymanager, llo-feeds, operatorforwarder, shared, vrf]
+        product:
+          # 98.5 is the aspirational code coverage once we migrate all tests to Foundry
+          # product that have that minimum coverage of 98.5 in the matrix below are currently excluded from the check
+          - name: automation
```

Comment:
Is it possible to move this to the top of the file, and also add the info if products have a gas snapshot, and if they use forge fmt? That would be a good single source of truth for all CI in this file

Reply:
this is a really cool idea!